.changeset/friendly-carpets-repeat.md

```diff
@@ -0,0 +1,45 @@
+---
+'@backstage/techdocs-common': minor
+'@backstage/plugin-techdocs': minor
+'@backstage/plugin-techdocs-backend': minor
+---
+
+_Breaking changes_
+
+1. Added option to use Google Cloud Storage as a choice to store the static generated files for TechDocs.
+   It can be configured using `techdocs.publisher.type` option in `app-config.yaml`.
+   Step-by-step guide to configure GCS is available here https://backstage.io/docs/features/techdocs/using-cloud-storage
+   Set `techdocs.publisher.type` to `'local'` if you want to continue using local filesystem to store TechDocs files.
+
+2. `techdocs.builder` is now required and can be set to `'local'` or `'ci'`. (Set it to `'local'` for now, since CI/CD build
+   workflow for TechDocs will be available soon (in few weeks)).
+   If builder is set to 'local' and you open a TechDocs page, `techdocs-backend` will try to build the docs, publish to storage and
+   show the generated docs afterwords.
+   If builder is set to `'ci'`, `techdocs-backend` will only fetch the docs and will NOT try to build and publish. In this case of 'ci',
+   we assume that docs are being built in the CI/CD pipeline of the repository.
+   TechDocs will not assume a default value for `techdocs.builder`. It is better to explicitly define it in the `app-config.yaml`.
+
+3. When configuring TechDocs in your backend, there is a difference in how a new publisher is created.
+
+```
+---  const publisher = new LocalPublish(logger, discovery);
++++  const publisher = Publisher.fromConfig(config, logger, discovery);
+```
+
+Based on the config `techdocs.publisher.type`, the publisher could be either Local publisher or Google Cloud Storage publisher.
+
+4. `techdocs.storageUrl` is now a required config. Should be `http://localhost:7000/api/techdocs/static/docs` in most setups.
+
+5. Parts of `@backstage/plugin-techdocs-backend` have been moved to a new package `@backstage/techdocs-common` to build docs. Also to publish docs
+   to-and-fro between TechDocs and a storage (either local or external). However, a Backstage app does NOT need to import the `techdocs-common` package -
```

Comment:
to-and-from/for?

Reply:
I could replace "to-and-fro" with "back and forth" :)

What I meant here is that there is a bidirectional communication between storage and TechDocs. And the new package is responsible for it.

1. Publishing generated docs to storage.
2. Retrieving generated docs and rending in Backstage/TechDocs.

Reply:
Didn't know that to-and-fro is actually an english term 😆 Yeah, maybe the "back and forth" is a bit simpler english 👍